### PR TITLE
docs: explain where to define variables for use in roles

### PR DESCRIPTION
**Pull request by andrewschoen**
As always, grammar nazis welcome and appreciated.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -44,6 +44,25 @@ if you prefer.
...
 
+Where should I put variables?
+-----------------------------
+
+All variables should be defined in ``defaults/main.yml`` for the role they're primarily used in.
+If the variable you're adding can be used in multiple roles define it in ``defaults/main.yml``
+for both roles. If the variable can contain a reasonable default value that should work for all
+possible labs then define that value in ``defaults/main.yml`` as well.  If not, you should still
+default the variable to something, but make the tasks that use the variable either fail
+gracefully without that var or prompt the user to define it if it's mandatory.
```
Comment (zmc) [inline README.rst]: Instead of failing at all, it seems we should just skip the task, no?

Comment (andrewschoen) [inline README.rst]: Depends on what the task is doing.  In some cases that task might be mandatory for the rest of the playbook to complete.

**Comment by andrewschoen** [comment]
Anybody have any feedback on this?  @dmick ?

File: README.rst [hunk 2]
```diff
@@ -44,6 +44,25 @@ if you prefer.
...
+
+If the variable is something that might need to be defined with a value specific to the lab in use, then
+it'll need to be added to your ``secrets`` repo as well. Variables in ``ansible/inventory/group_vars/all.yml`` will
```
Comment (zmc) [inline README.rst]: Could you rewrap this paragraph? It looks all wonky.

Comment (andrewschoen) [inline README.rst]: Sure, this might just be the github diff view doing that though.  I'll check.

Comment (andrewschoen) [inline README.rst]: @zmc I've rewrapped this, let me know if that looks better to you.

**Pull request merged**